Base/src/main/java/io/deephaven/base/FileUtils.java

```diff
@@ -282,21 +282,30 @@ public static URI convertToURI(final String source, final boolean isDirectory) {
         URI uri;
         try {
             uri = new URI(source);
+            if (uri.getScheme() == null) {
+                // Convert to a "file" URI
+                return convertToURI(new File(source), isDirectory);
+            }
+            String path = uri.getPath();
+            boolean isUpdated = false;
+            // Directory URIs should end with a slash
+            if (isDirectory && path.charAt(path.length() - 1) != URI_SEPARATOR_CHAR) {
+                path = path + URI_SEPARATOR_CHAR;
+                isUpdated = true;
+            }
```

Comment:
We should probably add some unit tests for the behavior we expect for convertToURI. Would probably be easy to break this out into a separate PR?

Reply:
So the new parquet S3 unit tests do test this code path, but you are right, I can add some dedicated tests for convertToURI and move it to a separate PR.